tests/integration/embed/embed_test.go

```diff
@@ -206,11 +206,15 @@ func setupEmbedCfg(cfg *embed.Config, curls []url.URL, purls []url.URL) {
 	cfg.ClusterState = "new"
 	cfg.ListenClientUrls, cfg.AdvertiseClientUrls = curls, curls
 	cfg.ListenPeerUrls, cfg.AdvertisePeerUrls = purls, purls
-	cfg.InitialCluster = ""
-	for i := range purls {
-		cfg.InitialCluster += ",default=" + purls[i].String()
+
+	if len(purls) > 0 {
+		builder := new(strings.Builder)
+		builder.WriteString(fmt.Sprintf("default=%s", purls[0].String()))
```

Comment:
I don't think the `url.URL` needs explicit casting into a string. The following should have the same result.

```golang
fmt.Sprintf("default=%s", purls[0])
```

Reply:
```go
fmt.Sprintf("default=%s", &purls[0]
```
I think you mean `&purls[0]`
```go
go doc url.URL.String
package url // import "net/url"

func (u *URL) String() string
```
Thanks, but I think it's just fine to call String() here and below.

Reply:
Good point; I didn't check the type of `purls`.